### PR TITLE
Simon bugfixes

### PR DESCRIPTION
**Pull request by dupreesi**
relates #424 #423 #420 #417 #274 

to test mailchimp u need to be in production mode

### CODE REVIEW

File: client/src/Components/Common/Button.style.js [hunk 1]
```diff
@@ -22,11 +22,14 @@ export const Button = styled.button`
...
   }
-
-  &:hover {
-    background-color: var(--secondary);
-    color: var(--white);
-  }
+  ${props => {
+    if (!props.disabled) {
+      return ` &:hover {
+        background-color: var(--secondary);
```
Comment (RamyAlshurafa) [inline client/src/Components/Common/Button.style.js]: use the theme context provider so we can get rid of global style with time

File: server/controllers/user/signup.js [hunk 2]
```diff
@@ -10,22 +10,40 @@ const {
...
-  const { email: _email, referredBy } = userInfo;
+  const { email: _email } = userInfo;
   const email = _email.toLowerCase();
 
   // check if email already exists
   try {
     const storedUser = await findByEmail(email);
     if (storedUser) {
-      return next(boom.conflict('Email already taken'));
+      next(boom.conflict('Email already taken'));
```
Comment (RamyAlshurafa) [inline server/controllers/user/signup.js]: you need a return here

File: server/controllers/user/signup.js [hunk 3]
```diff
@@ -10,22 +10,40 @@ const {
...
+
+    // ! PRODUCTION FEATURES
+    if (isProduction()) {
+      // add to mailchimp
+      if (user.mailingList) {
+        const addedToNewsletter = await addToMailChimp(email);
+        const { data } = addedToNewsletter;
+
+        if (data.errors && data.errors.length) {
+          next(boom.badData('error adding user to mailchimp'));
```
Comment (RamyAlshurafa) [inline server/controllers/user/signup.js]: need a "return" here also

File: client/src/Components/Common/Button.style.js (hunk 1 above)
Comment (dupreesi) [reply client/src/Components/Common/Button.style.js]: @ramy I can't get this working :/

File: client/src/Components/Common/Button.style.js [hunk 4]
```diff
@@ -20,13 +20,15 @@ export const Button = styled.button`
...
   }
 
-  &:hover {
-    background-color: var(--secondary);
-    color: var(--white);
-  }
+  ${props =>
+    !props.disabled &&
+    ` &:hover {
+      background-color: #e8529b;
```
Comment (RamyAlshurafa) [inline client/src/Components/Common/Button.style.js]:
```suggestion
      background-color: ${props.theme.colors.pink};
```

you can get it to work like this

**Pull request merged**